File: example/algorithms/02_getAll.py
import sys
import json

inpt = sys.stdin.readline()
inpt = json.loads(inpt)
docs = inpt["docs"]
q = inpt["q"]
opt = inpt["opt"]

result = []
for doc in docs:
    if doc["geotag"] is not None:
        result.append({"url":doc["url"], "tags":doc["tags"], "geotag":doc["geotag"]})

rtn = {"type":"map", "result":result, "q":q, "opt":opt}
print(json.dumps(rtn))

sys.exit(0)

File: example/algorithms/02_getAll.js
module.exports = (q, opt, docs, callback) => {
    opt.map = opt.map ? (opt.map == 'map' ? 'map' : 'heat') : 'map';
    let result = [];
    docs.forEach((p) => {
        if (p.geotag != null) {
            result.push({ url: p.url, tags: p.tags, geotag: p.geotag });
        }
    });
    callback(null, {
        'type': opt.map, //or 'scatter'
        'result': result,
        'q': q,
        'opt': opt
    });
}